clang/include/clang/3C/Duplicate.h

```diff
@@ -0,0 +1,24 @@
+#ifndef LLVM_DUPLICATE_H
+#define LLVM_DUPLICATE_H
+
+#include "ProgramInfo.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/Decl.h"
+#include "clang/AST/Stmt.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include <getopt.h>
+
+typedef std::function<bool (clang::VarDecl*)> selector;
+
+// Create a duplicate of a variable.
+// ToScan is the context to search for the variable
+// P is a predicate that decides which variable to make a duplicate of
+// Returns the name of the newly created variable
+// or None if no variable matching P is found
+Option<std::string> createDuplicate(clang::ASTContext &Context,
+                                    clang::Rewriter &R, ProgramInfo &Info,
+                                    clang::Decl *ToScan, selector P);
```

Comment:
I am wondering: Why this interface for determining the variable to duplicate? That is, I could imagine just specifying a single Decl and duplicating that. Or I could imagine providing a variable name, and you'd search for the proper decl. Just curious.

Reply:
This just seemed like the most general interface in case this becomes useful for other modules down the line. One change could be adding some top level helpers for simpler functionality.